src/code42cli/cmds/alerts.py

```diff
@@ -242,21 +234,75 @@ def search(
     **kwargs,
 ):
     """Search for alerts."""
-    output_header = ext.try_get_default_header(
+    output_header = try_get_default_header(
         include_all, _get_default_output_header(), format
     )
     formatter = OutputFormatter(format, output_header)
     cursor = _get_alert_cursor_store(cli_state.profile.name) if use_checkpoint else None
-    handlers = ext.create_handlers(
-        cli_state.sdk,
-        AlertExtractor,
-        cursor,
-        use_checkpoint,
-        formatter=formatter,
-        force_pager=include_all,
-    )
-    _call_extractor(cli_state, handlers, begin, end, or_query, advanced_query, **kwargs)
-    handle_no_events(not handlers.TOTAL_EVENTS and not errors.ERRORED)
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        checkpoint = cursor.get(checkpoint_name)
+        if checkpoint is not None:
+            begin = checkpoint
+
+    query = _construct_query(cli_state, begin, end, advanced_query, or_query)
+    alerts_list = _get_all_alerts(cli_state, query)
+
+    if not alerts_list:
+        click.echo("No results found.")
+        return
+
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        # update checkpoint to alertId of last event retrieved
+        alerts_gen = _store_updated_checkpoint(cursor, checkpoint_name, alerts_list)
+        formatter.echo_formatted_generated_output(alerts_gen)
```

Comment:
the `audit-logs` command was broken because this method didn't account for the `JSON/RAW-JSON/CSV` formats, so I removed it altogether. But I realize here we're going to have the same problem that there's currently not a good way to handle a generator of events for the non-Table formats... 

Maybe I should invest the time and try to sort that out now.

Reply:
I think we'll eventually be able to make the csv, and json format types handle generators, but I don't think table will ever really work with a generator, since you need to know what the max length of each column value is to get the spacing of the table right, and for that you need to have all the data up front.

It still is useful to use a generator for send-to though, since we're actually processing each event one at a time.

So just calling `list` on the generator before printing in `search` is fine for now.